encoder.go

```diff
@@ -56,6 +56,19 @@ const (
 	ObjectTooDeep
 )
 
+func (reason TruncationReason) String() string {
+	switch reason {
+	case ObjectTooDeep:
+		return "depth"
+	case ContainerTooLarge:
+		return "container-size"
+	case StringTooLong:
+		return "string-size"
+	default:
+		return ""
```

Comment:
Not a fan of that kind of `Stringer` returning blank... Either this returns something "somewhat helpful, albeit not pretty", or it panics.

```suggestion
		return fmt.Sprintf("TruncationReason<0x%02X>", reason)
```

Reply:
Panic is not really an option... going with your suggestion

Reply:
![image](https://github.com/DataDog/go-libddwaf/assets/47679741/85ba2ea2-81d1-4d7f-99f6-2bffabe158b8)